lib/rouge/lexers/liquid.rb

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Liquid < RegexLexer
+      title "Liquid"
+      desc 'Liquid is a templating engine for Ruby (liquidmarkup.org)'
+      tag 'liquid'
+      filenames '*.liquid'
+
+      def test(val)
+        mixin :generic
+      end
+
+      state :root do
+        rule /[^\{]+/, Text
+        rule /(\{%)(\s*)/ do
+          groups Punctuation, Text::Whitespace
+          push :tagBlock
+        end
+        rule /(\{\{)(\s*)/ do
+          groups Punctuation, Text::Whitespace
+          push :output
+        end
+        rule /\{/, Text
+      end
+
+      state :tagBlock do
+        # builtin logic blocks
+        rule /(if|unless|elsif|case)(?=\s+)/, Keyword::Reserved, :condition
+        rule /(when)(\s+)/ do
+          groups Keyword::Reserved, Text::Whitespace
+          push :when
+        end
+        rule (/(else)(\s*)(%\})/) do
+          groups Keyword::Reserved, Text::Whitespace, Punctuation
+          pop!
+        end
+
+        # other builtin blocks
+        rule (/(capture)(\s+)([^\s%]+)(\s*)(%\})/) do
+          groups Name::Tag, Text::Whitespace, Name::Attribute, Text::Whitespace, Punctuation
+          pop!
+        end
+        rule (/(comment)(\s*)(%\})/) do
+          groups Name::Tag, Text::Whitespace, Punctuation
+          push :comment
+        end
+        rule (/(raw)(\s*)(%\})/) do
+          groups Name::Tag, Text::Whitespace, Punctuation
+          push :raw
+        end
+        rule (/(assign)(\s*)/) do
+          groups Name::Tag, Text::Whitespace
+          push :assign
+        end
+        rule (/(include)(\s*)/) do
```

Comment:
Is there a reason you have to add `\s*` to the end here?  It's generally a little nicer just to mixin `:whitespace` to whatever state is next imo.